tools/test/audio/asrc_run.sh

```diff
@@ -2,25 +2,65 @@
 # SPDX-License-Identifier: BSD-3-Clause
 # Copyright(c) 2020 Intel Corporation. All rights reserved.
 
+# stop on most errors
+set -e
+
 usage ()
 {
-    echo "Usage:   $1 <bits in> <bits out> <rate in> <rate out> <input> <output>"
-    echo "Example: $1 16 16 32000 48000 input.raw output.raw"
+    echo "Usage:   $0 <options> <input> <output>"
+    echo "-c channels, default 2"
+    echo "-b bits_in, default 16"
+    echo "-r rate_in, default 48000"
+    echo "-R rate_out, default 48000"
+    echo "-v, for valgrind run"
+    echo "Example: $0 -r 32000 -R 48000 input.raw output.raw"
 }
 
 main()
 {
-    local COMP DIRECTION
+    local COMP DIRECTION CHANNELS BITS_IN BITS_OUT RATE VALGRIND
+    COMP=asrc
+    DIRECTION=playback
+    BITS="-b 16"
+    CHANNELS="-c 2"
+    RATE_IN="-r 48000"
+    RATE_OUT="-R 48000"
+    VALGRIND=
+    while getopts ":hvc:b:r:R:" opt; do
+	case "${opt}" in
+	    b)
+		BITS="-b ${OPTARG}"
+		;;
+	    c)
+		CHANNELS="-c ${OPTARG}"
+		;;
+	    h)
+		usage
+		exit
+		;;
+	    r)
+		RATE_IN="-r ${OPTARG}"
+		;;
+	    R)
+		RATE_OUT="-R ${OPTARG}"
+		;;
+	    v)
+		VALGRIND=-v
+		;;
+	    *)
+		usage
+		exit 1
+		;;
+	esac
+    done
 
-    if [ $# -ne 6 ]; then
+    shift $((OPTIND -1))
+    if [ $# -ne 2 ]; then
 	usage "$0"
-	exit
+	exit 1
     fi
 
-    COMP=asrc
-    DIRECTION=playback
-
-    ./comp_run.sh $COMP $DIRECTION "$@"
+    ./comp_run.sh $VALGRIND $BITS $CHANNELS $RATE_IN $RATE_OUT $COMP $DIRECTION "$@"
```

Comment:
You don't need to parse arguments in the first script where they are not used to merely pass them to the next script where they will be parsed again. That's a ton of duplication. As long as some order is enforced, the first script can instead use `shift` to remove its own arguments and then just pass-through the rest using `"$@"`

"$@" examples:
https://github.com/thesofproject/sof/blob/b95d995/scripts/docker-run.sh
https://github.com/thesofproject/sof/blob/b95d995bd2fb1f4f/scripts/xtensa-build-zephyr.sh#L385

and more in git grep -C 5 -F '"$@"'

But again: switching to config files and a single layer is much better.

Reply:
Thanks for tips, yep now using config files.